test/unit/publishProjectWebViewController.test.ts

```diff
@@ -53,4 +58,268 @@ suite("PublishProjectWebViewController Tests", () => {
+        const controller = new PublishProjectWebViewController(
+            contextStub,
+            vscodeWrapperStub,
+            projectPath,
+        );
+
+        await controller.initialized.promise;
+
+        // Set publish target to EXISTING_SERVER (default)
+        controller.state.formState.publishTarget = PublishTarget.ExistingServer;
+
+        const activeComponents = controller["getActiveFormComponents"](controller.state);
+
+        // Should include basic fields but NOT container fields
+        expect(activeComponents).to.include("publishTarget");
+        expect(activeComponents).to.include("publishProfilePath");
+        expect(activeComponents).to.include("serverName");
+        expect(activeComponents).to.include("databaseName");
+
+        // Should NOT include container fields
+        expect(activeComponents).to.not.include("containerPort");
+        expect(activeComponents).to.not.include("containerAdminPassword");
+    });
+
+    test("getActiveFormComponents returns correct fields for LOCAL_CONTAINER target", async () => {
+        const projectPath = "c:/work/TestProject.sqlproj";
+        const controller = new PublishProjectWebViewController(
+            contextStub,
+            vscodeWrapperStub,
+            projectPath,
+        );
+
+        await controller.initialized.promise;
+
+        // Set publish target to LOCAL_CONTAINER
+        controller.state.formState.publishTarget = PublishTarget.LocalContainer;
+
+        const activeComponents = controller["getActiveFormComponents"](controller.state);
+
+        // Should include basic fields AND container fields
+        expect(activeComponents).to.include("publishTarget");
+        expect(activeComponents).to.include("publishProfilePath");
+        expect(activeComponents).to.include("databaseName");
+
+        // Should include container fields
+        expect(activeComponents).to.include("containerPort");
+        expect(activeComponents).to.include("containerAdminPassword");
+        expect(activeComponents).to.include("containerAdminPasswordConfirm");
+        expect(activeComponents).to.include("containerImageTag");
+        expect(activeComponents).to.include("acceptContainerLicense");
+    });
+
+    test("state tracks inProgress and lastPublishResult", async () => {
+        const projectPath = "c:/work/TestProject.sqlproj";
+        const controller = new PublishProjectWebViewController(
+            contextStub,
+            vscodeWrapperStub,
+            projectPath,
+        );
+
+        await controller.initialized.promise;
+
+        // Initial state
+        expect(controller.state.inProgress).to.be.false;
+        expect(controller.state.lastPublishResult).to.be.undefined;
+
+        // Can be updated
+        controller.state.inProgress = true;
+        expect(controller.state.inProgress).to.be.true;
+    });
+
+    test("container target values are properly saved to formState", async () => {
```

Comment:
I know this duplication of formAction logic is temporary until we can convert the old stuff to the new one, but surely we just need to test one case here, since they're all the same (except when `isAction` changes)